src/DIRAC/WorkloadManagementSystem/DB/JobDB.py

```diff
@@ -1967,7 +1967,11 @@ def setHeartBeatData(self, jobID, dynamicDataDict):
         # If HeartBeatTime is being set, set it...
         timeStamp = dynamicDataDict.pop("HeartBeatTime", None)
         if timeStamp:
-            req = "UPDATE Jobs SET HeartBeatTime=%s WHERE JobID=%s" % (timeStamp, e_jobID)
+            result = self._escapeString(timeStamp)
+            if not result["OK"]:
+                self.log.warn("Failed to escape string ", timeStamp)
```

Comment:
Need to leave the function here, otherwise `req` is no defined and 1981 gives an exception, right?